pkg/ioutils/readers_test.go

```diff
@@ -2,59 +2,59 @@ package ioutils // import "github.com/docker/docker/pkg/ioutils"
 
 import (
 	"context"
-	"fmt"
+	"errors"
 	"io"
 	"strings"
 	"testing"
+	"testing/iotest"
 	"time"
-
-	"gotest.tools/v3/assert"
-	is "gotest.tools/v3/assert/cmp"
 )
 
-// Implement io.Reader
-type errorReader struct{}
-
-func (r *errorReader) Read(p []byte) (int, error) {
-	return 0, fmt.Errorf("error reader always fail")
-}
-
 func TestReadCloserWrapperClose(t *testing.T) {
-	reader := strings.NewReader("A string reader")
-	wrapper := NewReadCloserWrapper(reader, func() error {
-		return fmt.Errorf("This will be called when closing")
+	const text = "hello world"
+	testErr := errors.New("this will be called when closing")
+	wrapper := NewReadCloserWrapper(strings.NewReader(text), func() error {
+		return testErr
 	})
-	err := wrapper.Close()
-	if err == nil || !strings.Contains(err.Error(), "This will be called when closing") {
-		t.Fatalf("readCloserWrapper should have call the anonymous func and thus, fail.")
+
+	buf := make([]byte, len(text))
+	_, err := wrapper.Read(buf)
```

Comment:
Shouldn't the error be checked? I'm pretty sure a linter would complain about the unread assignment. Also https://pkg.go.dev/io#ReadAll
```suggestion
	buf, err := io.ReadAll(wrapper)
	if err != nil {
		t.Errorf("io.ReadAll(wrapper) err = %v", err)
	}
```

If you're wondering about the error message in my suggested change: https://github.com/golang/go/wiki/TestComments#identify-the-function

Reply:
Ah, yes, I can use that.

w.r.t. the error message, I'm on the fence; I frequently run into error-messages that don't match the code anymore, and for failures, the file and line-number are 99% of the time more useful to find the failure than the error message.

In the end; if the test fails, you'd be opening the file where it failed, navigate to the line number, and at that moment you already see where the error comes from.